### PR TITLE
Include name attribute in GraphStageLogic toString

### PR DESCRIPTION
**Pull request by nitikagarw**
References #29852

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 1]
```diff
@@ -1583,6 +1583,10 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
     override def toString = s"SubSourceOutlet($name)"
   }
 
+  override def toString: String = {
+    attributes.get[Attributes.Name].map(attr => s"${getClass.toString}-${attr.n}").getOrElse(getClass.toString)
+  }
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1588]: Usually with a single expression method body we leave out the parenthesises.

**Review by johanandren — APPROVED** [review]
One tiny thing about style, LGTM except for that.

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala (hunk 1 above)
Comment (nitikagarw) [reply akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala]: Thanks, @johanandren. I have removed the parenthesis.

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 2]
```diff
@@ -1583,6 +1583,9 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
     override def toString = s"SubSourceOutlet($name)"
   }
 
+  override def toString: String =
+    attributes.get[Attributes.Name].map(attr => s"${getClass.toString}-${attr.n}").getOrElse(getClass.toString)
+
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1588]: Can we move it before the inner classes to the other methods?

**Review by ennru — APPROVED** [review]
LGTM, just a personal preference about where to put it.

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala (hunk 2 above)
Comment (nitikagarw) [reply akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1588]: Sure, How about [here](https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala#L1006)?

Comment (ennru) [reply akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1588]: You know what, this class is so incomprehensibly big anyway, just leave it where it is.

**Comment by ennru** [comment]
Thank you, this will be helpful!

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala (hunk 2 above)
Comment (nitikagarw) [reply akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1588]: I just did :)

**Pull request merged**